browser/brave_federated_learning/brave_federated_learning_service_factory.cc

```diff
@@ -0,0 +1,76 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_federated_learning/brave_federated_learning_service_factory.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "brave/browser/brave_browser_process.h"
+#include "brave/browser/profiles/profile_util.h"
+#include "brave/components/brave_federated_learning/brave_federated_learning_service.h"
+#include "chrome/browser/browser_process.h"
```

Comment:
I guess you either need this or brave_browser_process

Reply:
and you need a dep for this

Reply:
I think I need `browser_process` to get the `local_state`, or is there another way to get the local state from `brave_browser_process`?